test/controllers/unattended_controller_test.rb

```diff
@@ -669,4 +669,31 @@ class UnattendedControllerTest < ActionController::TestCase
       end
     end
   end
+
+  context 'when safemode rendering is disabled' do
+    let(:os) { operatingsystems(:redhat) }
+    let(:ptable) { FactoryBot.create(:ptable, operatingsystem_ids: [os.id]) }
+    let(:host) do
+      FactoryBot.create(:host,
+        :managed,
+        build: true,
+        operatingsystem: os,
+        ptable: ptable,
+        medium: os.media.first,
+        architecture: os.architectures.first)
+    end
+
+    setup do
+      Setting[:safemode_render] = false
+    end
+
+    context 'with safemode parameter' do
+      it 'renders template in safemode' do
+        Foreman::Renderer::UnsafeModeRenderer.expects(:render).never
+        Foreman::Renderer::SafeModeRenderer.expects(:render).once
+
+        get :host_template, params: { kind: :provision, mac: host.mac, safemode: true }
```

Comment:
Wait, we are not changing `unattended_controller` in this PR, how is possible that the `safemode` param is taken into account?

Reply:
we are changing the `safe_render` method from `Foreman::Controller::TemplateRendering`, which is used in the `UnattendedController`

Reply:
Yes sorry forgot to delete this one from review, figure it out :-)